### ISSUE TITLE
Websocket node can get stuck in CONNECTING state

### ISSUE THREAD

**Issue by AndreKR** [node-red/node-red#4432]
### Current Behavior

I have a websocket-in node in connect-to mode.

The device it connects to was down this morning and while the device is up again, the node still shows "disconnected".

I wanted to investigate the issue a bit, so I went looking at [the source](https://github.com/node-red/node-red/blob/master/packages/node_modules/%40node-red/nodes/core/network/22-websocket.js) and I found that there is indeed code for automatic reconnect.

Being a bit stumped I randomly tried adding an inject node and a websocket-out node that connects to the same websocket client. I deployed it (in "modified flows" mode) and when I clicked the inject node, I got this error:

```
An error occurred while sending:  Error: WebSocket is not open: readyState 0 (CONNECTING)
```

### Expected Behavior

Ideally the node would just run into a timeout and automatically try to reconnect.

### Environment

- Node-RED version: v3.1.0 (Docker image `nodered/node-red:3.1-debian` started Oct 16)
- Node.js version: v16.20.2
- npm version:  8.19.4
- Platform/OS: Raspberry Pi OS 64 bit + Docker
- Browser: Chrome 118

**Comment by CoreyJ87** [comment]
Did you figure out what the problem was here?

**Comment by AndreKR** [comment]
No. It still happens about once every two weeks, at which point I get a Grafana alert and I log into Node-RED and click "Restart flows". 🙈

**Comment by CoreyJ87** [comment]
Hmm. Interesting. Im using it to connect to WLED. I have like 4 instances of the websocket connect node setup. 3 of them work fine. I have one that just now wont connect period. Ive tested the websocket and its up and working correctly. All of the wled devices are on the same vlan. So theres no connection issue. It just gives me this error no matter how many times I try to fire off the connection

**Comment by AndreKR** [comment]
If those WLED devices are on your local LAN I'm assuming you don't use TLS? In that case, maybe you can use tcpdump or Wireshark to record a dump of the connection process, once for a working device and once for the non-working device.
I don't think it's the exact same issue though.

**Comment by CoreyJ87** [comment]
Good idea. I'll give that a shot. Interesting addition. I tried another new esp. New ip. Same thing. Im wondering if making a "new" node in nodered right now is borked from the recent updates. Obviously thats probably not why you're having issues but could still be related

**Comment by AndreKR** [comment]
> Im wondering if making a "new" websocket node in nodered right now is borked from the recent updates.

Compare an old one to a new one in your flows.json.

**Comment by CoreyJ87** [comment]
> > Im wondering if making a "new" websocket node in nodered right now is borked from the recent updates.
> 
> Compare an old one to a new one in your flows.json.

Nope. Exactly the same....hmm. This is so weird. HASS and everything see wled fine. Its purely just the websocket that doesnt work. Odd. Ill have to try the wireshark thing later. Thanks for the help regardless.

**Comment by knolleary** [comment]
Closing as part of a tidy-up of the issue tracker — this has been inactive for a long time and much has changed in Node-RED since. If it's still relevant on a current version, please open a fresh issue with up-to-date details and we'll take a look. Thanks for raising it.
